extension/json/src/functions/table_functions/json_scan.cpp

```diff
@@ -563,11 +563,14 @@ static JsonScanFormat autoDetect(main::ClientContext* context, const std::string
         auto next = std::min<uint64_t>(numTuplesRead, numRowsToDetect);
         yyjson_val *key = nullptr, *ele = nullptr;
         for (auto i = 0u; i < next; i++) {
-            const auto& val = localState.values[i];
-            auto objIter = yyjson_obj_iter_with(val);
+            auto* doc = localState.docs[i];
```

Comment:
You probably don't need to write `*` after auto keyword.

Reply:
Since the semantics of pointer/reference types are different from values I personally prefer to write `auto*` or `auto&` to be a bit more explicit about things